### PR TITLE
Enable `use_savepoints` by default for Doctrine DBAL

### PR DESCRIPTION
**Pull request by fabpot**
| Q             | A
| ------------- | ---
| License       | MIT
| Doc issue/PR  | symfony/symfony-docs#...

### CODE REVIEW

File: doctrine/doctrine-bundle/2.0/config/packages/doctrine.yaml [hunk 1]
```diff
@@ -5,6 +5,7 @@ doctrine:
         # IMPORTANT: You MUST configure your server version,
         # either here or in the DATABASE_URL env var (see .env file)
         #server_version: '16'
+        use_savepoints: true
```
[the remark this answers is not in the document]
Comment (dmaicher) [reply doctrine/doctrine-bundle/2.0/config/packages/doctrine.yaml:8]: But anyway adding this for all versions is also fine

**Comment by dmaicher** [comment]
@derrabus @fabpot this will however lead to deprecations on DBAL 4. See https://github.com/symfony/demo/pull/1494#issuecomment-1945549526

:confused:

**Comment by derrabus** [comment]
Can you open an issue for that in the DoctrineBundle repository? We should fix it there.

**Comment by dmaicher** [comment]
> Can you open an issue for that in the DoctrineBundle repository? We should fix it there.

Hm but how would you fix it on DoctrineBundle? I mean on DBAL 4 not setting `use_savepoints` at all seems the way to go, or? This does not cause any deprecation. Eventually we should deprecate and remove the `use_savepoints` option from the Bundle I guess.

**Comment by dmaicher** [comment]
Let's discuss it [here](https://github.com/doctrine/DoctrineBundle/issues/1758)

**Comment by derrabus** [comment]
> Hm but how would you fix it on DoctrineBundle?

`use_savepoints: true` should be a no-op on DBAL 4 while `use_savepoints: false` should error.

> I mean on DBAL 4 not setting `use_savepoints` at all seems the way to go, or?

Yes. But projects should be able to gradually upgrade which is why it should be fine to still define this option on DBAL 4, even if there's only one valid setting.

**Pull request merged**